src/lib.rs

```diff
@@ -114,27 +218,30 @@ impl VsockListener {
             )
         } < 0
         {
-            Err(Error::last_os_error())
+            Err(P::last_os_error())
         } else {
-            Ok(SockAddr::Vsock(VsockAddr(vsock_addr)))
+            Ok(SockAddr(vsock_addr))
         }
     }
 
     /// Create a new independently owned handle to the underlying socket.
-    pub fn try_clone(&self) -> Result<Self> {
-        Ok(self.clone())
+    pub fn try_clone(&self) -> Result<VsockListener<P>, Error> {
+        Ok(VsockListener {
+            socket: self.socket.clone(),
+            phantom: PhantomData::<P>::default(),
+        })
```

Comment:
Is there a difference with the original implementation?

Reply:
The original implementation requires `P` to also implement `Clone`. I didn't want to impose that limitation as it doesn't really make sense. Any implementation of `P` should only exist at the type system level, it shouldn't occupy any memory so what does it even mean to `Clone` something that has zero size?

Reply:
The type parameter in `PhantomData::<P>` isn't required though. Removed it

Reply:
For future reference: after some offline discussion we came to the conclusion that the current implementation should be kept. The `#[derive(Clone)]` on `VsockListener` generates a `Clone` implementation with a `Clone` type bound on `P`, which is too strict since `PhantomData<P>` implements `Clone` even if `P` doesn't.

There was also discussion on whether `Clone` and `try_clone()` even make sense for a `VsockListener`, but we decided to keep those, in order to keep the code as close as possible to the upstream version.